src/test/run-pass/consts/const-ptr-nonnull.rs

```diff
@@ -0,0 +1,22 @@
+// run-pass
+
+#![feature(const_ptr_nonnull)]
+
+use std::ptr::NonNull;
+
+const fn dangling() -> NonNull<u32> {
+    NonNull::dangling()
+}
+
+const fn cast<T, U>(non_null: NonNull<T>) -> NonNull<U> {
+    non_null.cast()
+}
+
+pub fn main() {
+    assert_eq!(dangling(), NonNull::dangling());
+
+    let mut i: i32 = 10;
+    let non_null_t = NonNull::new(&mut i).unwrap();
```

Comment:
While the above functions are `const`, the evaluation doesn't happen via const eval. In order to trigger const eval, you're going to need an explicit `const FOO` item somewhere.

Reply:
Should be fixed now

Reply:
thanks!